services/QuillLMS/engines/comprehension/app/models/comprehension/rule.rb

```diff
@@ -68,14 +68,19 @@ def determine_feedback_from_history(feedback_history)
 
     def regex_is_passing?(entry)
       regex_rules.none? do |regex_rule|
-        regex_rule.sequence_type == RegexRule::TYPE_INCORRECT ? Regexp.new(regex_rule.regex_text).match(entry) : !Regexp.new(regex_rule.regex_text).match(entry)
+        is_matching = match_with_case_sensitive(regex_rule.regex_text, regex_rule.case_sensitive, entry)
```

Comment:
It's hard to follow this method. `is_` by convention connotes a boolean, but in this case returns MatchData. Please rename the variable.

line 72 is also not clear to me, but perhaps fixing the variable name will help.